semantic-python/src/Language/Python/Core.hs

```diff
@@ -93,21 +122,30 @@ instance Compile (Py.AssertStatement Span)
 instance Compile (Py.Attribute Span)
 
 instance Compile (Py.Assignment Span) where
-  compile it@Py.Assignment { Py.left = Py.ExpressionList { Py.extraChildren = [lhs] }, Py.right = Just rhs } = do
-    target <- compile lhs
+  compileCC it@Py.Assignment
+    { Py.left = Py.ExpressionList
+      { Py.extraChildren =
+        [ Py.PrimaryExpressionExpression (Py.IdentifierPrimaryExpression (Py.Identifier { Py.bytes = name }))
+        ]
+      }
+    , Py.right = Just rhs
+    } cc = do
     value  <- compile rhs
-    locate it $ target .= value
-  compile other = fail ("Unhandled assignment case: " <> show other)
+    let assigning n = (Name.named' name :<- value) >>>= n
+    locate it =<< assigning <$> local (def name) cc
```

Comment:
I think this is the right choice for assignments to locals; for stuff like member accesses it might be more involved, but that’s ok.

Reiterating some of our conversation earlier today, this means that variables at the top-level are part of a module’s exported interface.